### ISSUE TITLE
Exception if field type is altered in non-atomic migration

### ISSUE THREAD

**Issue by ferdinand-hoffmann** [Opus10/django-pgtrigger#132]
```shell
$ ./manage.py migrate 
Operations to perform:
  Apply all migrations: [...snip...]
Running migrations:
  [...snip...]
  Applying wagtailcore.0070_rename_pagerevision_revision...Traceback (most recent call last):
  File "/code/./manage.py", line 46, in <module>
    main()
  File "/code/./manage.py", line 42, in main
    execute_from_command_line(sys.argv)
  File "/venv/lib/python3.11/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/venv/lib/python3.11/site-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/venv/lib/python3.11/site-packages/django/core/management/base.py", line 412, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/venv/lib/python3.11/site-packages/django/core/management/base.py", line 458, in execute
    output = self.handle(*args, **options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/django/core/management/base.py", line 106, in wrapper
    res = handle_func(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/django/core/management/commands/migrate.py", line 356, in handle
    post_migrate_state = executor.migrate(
                         ^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/django/db/migrations/executor.py", line 135, in migrate
    state = self._migrate_all_forwards(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/django/db/migrations/executor.py", line 167, in _migrate_all_forwards
    state = self.apply_migration(
            ^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/django/db/migrations/executor.py", line 252, in apply_migration
    state = migration.apply(state, schema_editor)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/django/db/migrations/migration.py", line 132, in apply
    operation.database_forwards(
  File "/venv/lib/python3.11/site-packages/django/db/migrations/operations/fields.py", line 235, in database_forwards
    schema_editor.alter_field(from_model, from_field, to_field)
  File "/venv/lib/python3.11/site-packages/django/db/backends/base/schema.py", line 830, in alter_field
    self._alter_field(
  File "/venv/lib/python3.11/site-packages/pgtrigger/migrations.py", line 336, in _alter_field
    return super()._alter_field(
           ^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/django/db/backends/postgresql/schema.py", line 287, in _alter_field
    super()._alter_field(
  File "/venv/lib/python3.11/site-packages/django/db/backends/base/schema.py", line 879, in _alter_field
    self.execute(self._delete_fk_sql(model, fk_name))
  File "/venv/lib/python3.11/site-packages/pgtrigger/migrations.py", line 380, in execute
    with self.atomic:
         ^^^^^^^^^^^
AttributeError: 'DatabaseSchemaEditor' object has no attribute 'atomic'
```

this happened with the following migration from wagtail: https://github.com/wagtail/wagtail/blob/main/wagtail/migrations/0070_rename_pagerevision_revision.py#L20

**Comment by Headmaster11** [comment]
Have the same error with Wagtail

**Comment by ferdinand-hoffmann** [comment]
Is it possible to replace `self.atomic` with `transaction.atomic` in `DatabaseSchemaEditorMixin`? This way the code would be independent from the outer transaction.

**Comment by kekekekule** [comment]
Tried to solve it here: https://github.com/Opus10/django-pgtrigger/pull/136

Welcome for a review

**Comment by wesleykendall** [comment]
@ferdinand-hoffmann that's how I ended up addressing it in https://github.com/Opus10/django-pgtrigger/pull/137

Thanks for letting me know and thanks for your contributions @kekekekule 

Fix deployed in 4.10.0!
